testing/python/metafunc.py

```diff
@@ -1503,6 +1548,31 @@ def test_it(x): pass
         result = pytester.runpytest()
         assert result.ret == 0
 
+    def test_parametrize_module_level_test_with_class_scope(
+        self, pytester: Pytester
+    ) -> None:
+        module = pytester.makepyfile(
+            """
+            import pytest
+
+            @pytest.mark.parametrize("x", [0, 1], scope="class")
+            def test_1(x):
+                pass
+
+            @pytest.mark.parametrize("x", [1, 2], scope="module")
+            def test_2(x):
+                pass
+        """
+        )
+        test_1_0, _, test_2_0, _ = pytester.genitems((pytester.getmodulecol(module),))
```

Comment:
Let's add asserts that `test_1_0` and `test_2_0` are actually those, so that in case the ordering becomes wrong for some reason then we'll fail.